packages/teleterm/src/ui/TabHost/useTabShortcuts.ts

```diff
@@ -25,7 +25,6 @@ import { useNewTabOpener } from 'teleterm/ui/TabHost/useNewTabOpener';
 
 export function useTabShortcuts() {
   const { workspacesService } = useAppContext();
-  workspacesService.useState();
```

Comment:
Ah, it looks like I didn't commit all the changes I wanted. 
I wanted to remove `workspacesService.useState()` from `TabHost` as well and leave it only in `TabHostContainer`.

I agree with you, we should be more explicit, I added whole `ctx` object to the `TabHost`, `useTabShortcuts`, `useNewTabOpener` (some of these need more than only `documentsServive`).

Reply:
I suggested passing `documentService` around since it was my impression that since `useTabShortcuts` is no longer subscribed to `workspacesService`, we could remove the dependency on `workspacesService` altogether and just pass `documentService` from outside.

If this was not done, we could run into problems where, for example, we take the logic from `useTabShortcuts` that depends on getting the active `documentService` and we move it outside of `TabHost`. Since `useTabShortcuts` would no longer be called again when the active `documentService` changes, this could lead to bugs.

You explained that it is not enough to pass `documentService`, which is fine. But now what does passing `ctx` give us? In theory it can still be given `ctx` by a component that doesn't listen to `workspacesService` updates, so we might as well call `useAppContext` there.

---

It seems like the only other piece of data that's needed is `localClusterUri`. In the future, it'd be helpful to have a single object that fully represents a workspace, that is it has access to both `localClusterUri` and the document service for that workspace. Then we could just pass this one single thing around.

Reply:
Totally agree, for now I added `{documentService, localClusterUri}` object as a param (but we should refactor `workspace/documentsService`).